docs/composition.ipynb

```diff
@@ -0,0 +1,739 @@
+    "input_space = dp.vector_domain(dp.atom_domain(T=dp.usize), 31), dp.linf_distance(T=dp.usize)\n",
+    "select_index_measurement = dp.m.make_report_noisy_max_gumbel(*input_space, scale=1.0, optimize='min')\n",
+    "\n",
+    "median_meas = discrete_scores >> select_index_measurement >> (lambda index: candidates[index])\n",
+    "print(\"DP Median:\", sc_qbl(median_meas))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "That was a lot of code! But as you saw, we could change our queries depending on the outputs of other queries, illustrating interactive sequential composition."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Interactive Sequential Composition with Polars Queries"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can also implement interactive sequential composition with polars queries! One of the first steps is to specify the domain of our dataframe and add a margin specifying the parameters for the data when it's grouped by `ILOSTAT`. This is **in addition** to specifying it in the compositor. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "DF Domain:  FrameDomain(COEFF: f64, QUARTER: i64, REFYEAR: i64, REFWEEK: i64, INTWEEK: f64, COUNTRY: str, DEGURBA: f64, HHINST: f64, INTWAVE: i64, INTQUEST: i64, REM: i64, YEAR: i64, HHPRIV: i64, SEX: i64, AGE: f64, NATIONAL: str, YEARESID: f64, COUNTRYB: str, PROXY: f64, NOWKREAS: i64, STAPRO: f64, SIGNISAL: i64, COUNTRYW: str, YSTARTWK: f64, MSTARTWK: f64, FTPT: f64, TEMP: f64, TEMPDUR: f64, HWUSUAL: f64, HWACTUAL: f64, HWOVERP: f64, HWOVERPU: f64, HOURREAS: f64, WISHMORE: f64, HWWISH: f64, LOOKOJ: i64, EXIST2J: f64, STAPRO2J: f64, HWACTUA2: f64, EXISTPR: f64, YEARPR: f64, MONTHPR: f64, STAPROPR: f64, SEEKWORK: i64, SEEKTYPE: f64, SEEKDUR: f64, METHODA: i64, METHODB: i64, METHODC: i64, METHODD: i64, METHODE: i64, METHODF: i64, METHODG: i64, METHODH: i64, METHODI: i64, METHODJ: i64, METHODK: i64, METHODL: i64, METHODM: i64, WANTWORK: f64, AVAILBLE: i64, EDUCSTAT: f64, EDUCLEVL: f64, COURATT: f64, COURLEN: f64, ILOSTAT: i64, ISCO1D: f64, ISCOPR1D: f64, DURUNE: f64, EDUC4WN: f64, HATLEV1D: str, STARTIME: f64, LEAVCLAS: f64, NACE1D: str, NACE2J1D: str, NACEPR1D: str, HHTYPE: i64; margins=[{\"ILOSTAT\"}])\n"
+     ]
+    }
+   ],
+   "source": [
+    "#get the domain of the datafarame\n",
+    "df_domain = dp.domain_of(df, infer=True)\n",
+    "#specify the parameters when grouped by Working cateory\n",
+    "df_domain = dp.with_margin(df_domain, by=[\"ILOSTAT\"], \n",
+    "                           public_info=\"lengths\", \n",
+    "                           max_partition_length=50, \n",
+    "                           max_num_partitions=3)\n",
+    "print(\"DF Domain: \",df_domain)\n"
```

Comment:
Generally, I do like to show the intermediate object... but here it's so big that I worry it would be a distraction to the user.